src/docfx/config/ops/OpsConfigAdapter.cs

```diff
@@ -141,13 +143,31 @@ await _opsAccessor.GetDocumentUrls(), new[] { new { log_code = "", document_url
                     $"{PublicMetadataSchemaApi}{metadataServiceQueryParams}",
                 },
                 allowlists = AllowlistsApi,
+                allowedHTML,
                 trustedDomains,
                 sandboxEnabledModuleList = SandboxEnabledModuleListApi,
                 xref = xrefMaps,
                 isReferenceRepository = docsets.Any(d => d.use_template),
             });
         }
 
+        public static Dictionary<string, HashSet<string>?> ConvertAllowedHTML(string json)
+        {
+            var taxonomies = JsonConvert.DeserializeObject<Taxonomies>(json) ?? new();
+            if (taxonomies.TryGetValue(AllowedHtml, out var taxonomy))
+            {
+                var taxoAllowedTags = taxonomy.NestedTaxonomy.dic
+                    .Select(item => (item.Key, Value: item.Value.Where(i => !"(empty)".Equals(i, StringComparison.OrdinalIgnoreCase)).ToArray()))
```

Comment:
@OsmondJiang should the taxonomy service be updated to not return `(empty)` at all?

Reply:
(empty) means the `key` is allowed to exist alone

Reply:
before moving the allow list to poolparty, we have a feature that key1 and key2 must be used together, for this case, （empty) is not allowed